webserver/webserver/src/main/java/io/helidon/webserver/HttpRoute.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2022 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.helidon.webserver;
+
+import java.util.List;
+
+import io.helidon.common.http.Http;
+
+/**
+ * A basic HTTP route (should be usable by ANY HTTP protocol version).
+ */
+public interface HttpRoute extends Route {
+
+    /**
+     * HTTP protocol version supported by the route.
+     *
+     * @param version HTTP protocol version supported by the route
+     * @return true if supported
+     */
+    default boolean matchVersion(Http.Version version) {
+        return true;
+    }
+
+    /**
+     * Create HTTP route.
+     *
+     * @param method  accepted method
+     * @param path    path pattern
+     * @param handler handler
+     * @return a new HTTP route
+     */
+    static HttpRoute route(Http.Method method, String path, Handler handler) {
```

Comment:
As I mentioned in Http1Route - this is a limited purpose method, we should provide a more generic one with `PathMatcher` and vararg of methods